corehq/apps/geospatial/views.py

```diff
@@ -163,3 +181,118 @@ def post(self, request, *args, **kwargs):
         instance.save()
 
         return self.get(request, *args, **kwargs)
+
+
+class GPSCaptureView(BaseDomainView):
+    urlname = 'gps_capture'
+    template_name = 'gps_capture.html'
+
+    page_name = _("Manage GPS Data")
+    section_name = _("Geospatial")
+
+    @method_decorator(toggles.GEOSPATIAL.required_decorator())
+    def dispatch(self, *args, **kwargs):
+        return super(GPSCaptureView, self).dispatch(*args, **kwargs)
+
+    @property
+    def section_url(self):
+        return reverse(self.urlname, args=(self.domain,))
+
+    @property
+    def page_url(self):
+        return reverse(self.urlname, args=(self.domain,))
+
+    @property
+    def page_context(self):
+        data_type = self.request.GET.get('data_type', 'case')
+        return {
+            'data_type': data_type
+        }
+
+    @method_decorator(toggles.GEOSPATIAL.required_decorator())
+    def post(self, request, *args, **kwargs):
+        json_data = json.loads(request.body)
+        data_type = json_data.get('data_type', None)
+        data_item = json_data.get('data_item', None)
+
+        if data_type == 'case':
+            process_gps_values_for_case(request.domain, data_item)
+        elif data_type == 'user':
+            process_gps_values_for_user(request.domain, data_item)
+
+        return json_response({
+            'status': 'success'
+        })
+
+
+@require_GET
+@login_and_domain_required
+def get_paginated_cases_or_users_without_gps(request, domain):
+    page = int(request.GET.get('page', 1))
+    limit = int(request.GET.get('limit', 5))
+    case_or_user = request.GET.get('data_type', 'case')
+    query = request.GET.get('query', '')
+    assert page > 0, 'Page must be a positive number'
+    assert limit > 0, 'Limit must be a positive number'
+    assert case_or_user == 'case' or case_or_user == 'user', 'Data type must be "case" or "user"'
```

Comment:
For the capture page, if these assertions managed to fail then a 500 will be returned and the user will see an error banner.

I agree this could be dealt with in a more elegant manner so as to not return a 500 back. Looking at other places in HQ (case imports for example) I see that the input parameters are given defaults, so I'll change this view to do the same.

Reply:
[af9af69](https://github.com/dimagi/commcare-hq/pull/33420/commits/af9af6978d4832a82d427b3dc7b0ba687a0d9b77)